### PR TITLE
Asks page

### PR DESCRIPTION
**Pull request by alonpeled44**

### CODE REVIEW

File: components/Ask.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import styles from "@/styles/components/ask.module.scss";
+import React, { useState } from "react";
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: server/asksData.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const asksData = {
+	"0": {ask: "Can a dog love me?", like: false, anonymous: true },
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply server/asksData.js]: noted

File: components/Ask.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import asksData from "@/server/asksData";
+
+export default function Ask({ id, ask, like, anonymous }) {
+	const [isLiked, setIsLike] = useState(like);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: server/asksData.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply server/asksData.js]: noted

File: components/Ask.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		setIsLike((prev) => {
+			asksData[id]["like"] = !prev;
+			return !prev;
+		});
+	};
+
+	return (
+		<article className={styles.ask}>
+			<h2>{ask}</h2>
+			<img src='/blob.png' alt='blob' />
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/Ask.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+			<img onClick={handleClicked} src={isLiked ? "/likeActive.png" : "/likeNotActive.png"} alt='like' />
+			{!anonymous && (
+				<a href={`/account/${id}`}>
+					<img src='/lio_img.png' />
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/Ask.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import styles from "@/styles/components/ask.module.scss";
+import React, { useState } from "react";
+import asksData from "@/server/asksData";
+
+export default function Ask({ id, ask, like, anonymous }) {
+	const [isLiked, setIsLike] = useState(like);
+
+	const handleClicked = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/Ask.jsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+		setIsLike((prev) => {
+			asksData[id]["like"] = !prev;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/Ask.jsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+			return !prev;
+		});
+	};
+
+	return (
+		<article className={styles.ask}>
+			<h2>{ask}</h2>
+			<img src='/blob.png' alt='blob' />
+			<img onClick={handleClicked} src={isLiked ? "/likeActive.png" : "/likeNotActive.png"} alt='like' />
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/Ask.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/Ask.jsx]: noted

File: components/AskForm.jsx [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import styles from "@/styles/components/ask-form.module.scss";
+import React, { useState } from "react";
+
+export default function AskForm({ setAskFormActive }) {
+	const [isMarked, setIsMarked] = useState(false);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx:5]: noted

File: components/AskForm.jsx [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+	const handleBackgroundClicked = () => {
+		setAskFormActive(false);
+	};
+
+	return (
+		<div className={styles["form-pop-up"]}>
+			<div className={styles.background} onClick={handleBackgroundClicked} />
+			<form className={styles["ask-form"]}>
+				<h2>Ask whatever!</h2>
+				<div className={styles["user-information"]}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+					<input type='text' placeholder='Type here...' />
+					<label htmlFor='anonymous'>
+						<input type='checkbox' name='anonymous' id='anonymous' onClick={handleCheckboxClicked} />
+						<span>{isMarked ? "✓" : null}</span>
+						<span>Anonymous</span>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+	return (
+		<div className={styles["form-pop-up"]}>
+			<div className={styles.background} onClick={handleBackgroundClicked} />
+			<form className={styles["ask-form"]}>
+				<h2>Ask whatever!</h2>
+				<div className={styles["user-information"]}>
+					<input type='text' placeholder='Type here...' />
+					<label htmlFor='anonymous'>
+						<input type='checkbox' name='anonymous' id='anonymous' onClick={handleCheckboxClicked} />
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+						<span>{isMarked ? "✓" : null}</span>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+	};
+
+	const handleBackgroundClicked = () => {
+		setAskFormActive(false);
+	};
+
+	return (
+		<div className={styles["form-pop-up"]}>
+			<div className={styles.background} onClick={handleBackgroundClicked} />
+			<form className={styles["ask-form"]}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 15]
```diff
@@ -0,0 +1,34 @@
+import styles from "@/styles/components/ask-form.module.scss";
+import React, { useState } from "react";
+
+export default function AskForm({ setAskFormActive }) {
+	const [isMarked, setIsMarked] = useState(false);
+
+	const handleCheckboxClicked = () => {
+		setIsMarked((prev) => {
+			return !prev;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx:9]: noted

File: components/AskForm.jsx [hunk 16]
```diff
@@ -0,0 +1,34 @@
+import styles from "@/styles/components/ask-form.module.scss";
+import React, { useState } from "react";
+
+export default function AskForm({ setAskFormActive }) {
+	const [isMarked, setIsMarked] = useState(false);
+
+	const handleCheckboxClicked = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx:7]: noted

File: components/AskForm.jsx [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+		setIsMarked((prev) => {
+			return !prev;
+		});
+	};
+
+	const handleBackgroundClicked = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx:13]: noted

File: pages/asks.jsx [hunk 18]
```diff
@@ -0,0 +1,64 @@
+import styles from "@/styles/pages/asks.module.scss";
+import Ask from "@/components/Ask";
+import asksData from "@/server/asksData";
+import AskForm from "@/components/AskForm";
+import React, { useState } from "react";
+
+export default function Asks() {
+	const askDataArr = Object.entries(asksData);
+	const [likedActive, setLikedActive] = useState(false);
+	const [askFormActive, setAskFormActive] = useState(false);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: components/AskForm.jsx
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx:1]: noted

File: pages/asks.jsx [hunk 19]
```diff
@@ -0,0 +1,64 @@
+import styles from "@/styles/pages/asks.module.scss";
+import Ask from "@/components/Ask";
+import asksData from "@/server/asksData";
+import AskForm from "@/components/AskForm";
+import React, { useState } from "react";
+
+export default function Asks() {
+	const askDataArr = Object.entries(asksData);
+	const [likedActive, setLikedActive] = useState(false);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx:9]: noted

File: pages/asks.jsx [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+	const [askFormActive, setAskFormActive] = useState(false);
+
+	const handleAskClick = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx:12]: noted

File: pages/asks.jsx [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+		setAskFormActive((prev) => {
+			return !prev;
+		});
+	};
+
+	const handleLikeButtonClick = () => {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: pages/asks.jsx [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+		setLikedActive((prev) => {
+			return !prev;
+		});
+	};
+	let showAsk = [];
+
+	if (likedActive) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: pages/asks.jsx [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+			return !prev;
+		});
+	};
+
+	const handleLikeButtonClick = () => {
+		setLikedActive((prev) => {
+			return !prev;
+		});
+	};
+	let showAsk = [];
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: pages/asks.jsx [hunk 24]
```diff
@@ -0,0 +1,64 @@
...
+
+	if (likedActive) {
+		const likedAsks = askDataArr.filter((dataArr) => dataArr[1]["like"] === true);
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: server/asksData.js [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+	"1": {ask: "Can a dog love me?", like: true, anonymous: true },
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply server/asksData.js]: noted

File: styles/components/ask-form.module.scss [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+				font-size: 1.4rem;
+
+				&::placeholder {
+					color: #c8c8c8;
+				}
+			}
+
+			label {
+				display: flex;
+				justify-content: flex-start;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

File: styles/components/ask-form.module.scss [hunk 27]
```diff
@@ -0,0 +1,158 @@
...
+				align-items: center;
+				width: fit-content;
+				gap: 1.75rem;
+
+				input[type="checkbox"] {
+					display: none;
+
+					& + span {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

File: styles/components/ask-form.module.scss [hunk 28]
```diff
@@ -0,0 +1,158 @@
...
+						align-items: center;
+						border: colors.$secondary 0.3rem solid;
+						border-radius: 0.75rem;
+						width: 2.75rem;
+						aspect-ratio: 1/1;
+						font-weight: 800;
+					}
+				}
+
+				span {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

File: styles/components/ask-form.module.scss [hunk 29]
```diff
@@ -0,0 +1,158 @@
...
+					height: 3rem;
+					padding: 1rem;
+					font-size: 1rem;
+				}
+
+				label {
+					gap: 1.25rem;
+
+					input[type="checkbox"] {
+						& + span {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

File: styles/components/ask.module.scss [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+	h2 {
+		font-size: 1.7rem;
+		font-weight: 400;
+		user-select: none;
+	}
+
+	img {
+		position: absolute;
+
+		&:nth-child(2) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask.module.scss]: noted

File: styles/components/ask.module.scss [hunk 31]
```diff
@@ -0,0 +1,52 @@
...
+	align-items: center;
+
+	h2 {
+		font-size: 1.7rem;
+		font-weight: 400;
+		user-select: none;
+	}
+
+	img {
+		position: absolute;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask.module.scss:21]: noted

File: styles/components/ask.module.scss [hunk 32]
```diff
@@ -0,0 +1,52 @@
...
+			width: 1.5rem;
+		}
+
+		&:nth-child(3) {
+			left: 0.65rem;
+			bottom: 0.5rem;
+			width: 1.2rem;
+		}
+	}
+	a {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask.module.scss:34]: noted

File: styles/components/ask.module.scss [hunk 33]
```diff
@@ -0,0 +1,52 @@
...
+			left: -0.5rem;
+			border-radius: 50%;
+		}
+	}
+}
+
+@media only screen and (max-width: 800px) {
+	.ask {
+		width: 100%;
+		h2 {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask.module.scss:52]: noted

File: styles/pages/asks.module.scss [hunk 34]
```diff
@@ -0,0 +1,88 @@
+@use "/styles/colors";
+
+.asks-and-buttons {
+	height: 100%;
+	box-sizing: border-box;
+	padding-top: 2rem;
+	display: flex;
+	flex-direction: column;
+	align-items: center;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/pages/asks.module.scss]: noted

File: styles/pages/asks.module.scss
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/pages/asks.module.scss:1]: noted

File: styles/pages/asks.module.scss [hunk 35]
```diff
@@ -0,0 +1,88 @@
...
+				color: colors.$primary;
+				background-color: colors.$secondary;
+			}
+		}
+	}
+}
+
+@media only screen and (max-width: 800px) {
+	.asks-and-buttons {
+		padding-top: 1rem;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/pages/asks.module.scss]: noted

File: styles/pages/asks.module.scss [hunk 36]
```diff
@@ -0,0 +1,88 @@
...
+		padding-bottom: 1.5rem;
+		padding-right: 1.5rem;
+		padding-left: 1rem;
+		.asks {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/pages/asks.module.scss:71]: noted

File: styles/pages/asks.module.scss [hunk 37]
```diff
@@ -0,0 +1,88 @@
...
+		button {
+			padding: 1rem;
+			min-width: 12rem;
+			box-sizing: border-box;
+			border-radius: 0.5rem;
+			border: solid 0.2rem colors.$secondary;
+			font-size: 1.5rem;
+			font-weight: 600;
+
+			&:nth-child(1) {
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/pages/asks.module.scss]: noted

File: pages/asks.jsx [hunk 38]
```diff
@@ -0,0 +1,64 @@
...
+					like={dataArr[1]["like"]}
+					anonymous={dataArr[1]["anonymous"]}
+				/>
+			);
+		});
+	}
+
+	return (
+		<main>
+			<div className={styles["asks-and-buttons"]}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: styles/components/ask-form.module.scss [hunk 39]
```diff
@@ -0,0 +1,158 @@
...
+
+	.background {
+		height: 100vh;
+		width: 100vw;
+		background-color: rgba($color: #000000, $alpha: 0.5);
+		backdrop-filter: blur(5px);
+	}
+
+	.ask-form {
+		position: absolute;
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

[the remark this answers is not in the document]
Comment (alonpeled44) [reply styles/components/ask-form.module.scss]: noted

File: components/AskForm.jsx [hunk 40]
```diff
@@ -0,0 +1,34 @@
...
+		setAskFormActive(false);
+	};
+
+	return (
+		<div className={styles["form-pop-up"]}>
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: components/AskForm.jsx [hunk 41]
```diff
@@ -0,0 +1,34 @@
...
+			<div className={styles.background} onClick={handleBackgroundClicked} />
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply components/AskForm.jsx]: noted

File: pages/asks.jsx
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx:1]: noted

File: components/Modal.jsx [hunk 42]
```diff
@@ -0,0 +1,34 @@
...
+export default function Modal({ content, isModalOpen, setIsModalOpen }) {
+	const modalRef = useRef(null);
+
+	useEffect(() => {
+		if (isModalOpen) {
+			modalRef.current.showModal();
+		}
+	}, [isModalOpen]);
+
+	const backgroundClicked = (event) => {
```
Comment (liogiladi) [inline components/Modal.jsx:13]: think it would be better to call it `handleBackgroundClick` as it would be more consistent with other events you called before.

File: components/Modal.jsx [hunk 43]
```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect } from "react";
+import styles from "@/styles/components/modal.module.scss";
+
+export default function Modal({ content, isModalOpen, setIsModalOpen }) {
```
Comment (liogiladi) [inline components/Modal.jsx:4]: Use `children` instead of `content`. There is more than semantical aspect to it - 
https://react.dev/learn/passing-props-to-a-component#passing-jsx-as-children

Comment (liogiladi) [inline components/Modal.jsx:4]: `isModalOpen` is redundant as you are already in Modal context. Call it `isOpen` instead. The same with the setter.

File: components/Modal.jsx [hunk 44]
```diff
@@ -0,0 +1,34 @@
...
+	const modalRef = useRef(null);
```
Comment (liogiladi) [inline components/Modal.jsx:5]: Suggest calling it `ref` instead of `modalRef` as it's a ref to the parent element of the component and the only ref.
On the other side, one can make the argument that it's more clear to define in-name to which the ref belongs.
You choice.

File: pages/asks.jsx [hunk 45]
```diff
@@ -0,0 +1,54 @@
+import { useState, useMemo } from "react";
+import styles from "@/styles/pages/asks.module.scss";
+import Ask from "@/components/Ask";
+import content from "@/server/content";
+import CreateAskForm from "@/components/CreateAskForm";
+import Modal from "@/components/Modal";
+
+export default function Asks() {
+	const contentArr = Object.entries(content);
```
Comment (liogiladi) [inline pages/asks.jsx:9]: Commented before, splitting the `Object.entries` from the calculation of the list's elements introduce more confusion than clarity as its purpose is separated from the implementation. Code units in general should be colocated to other related units.

* Also it bring an hindrance on performance that you tried to prevent. 
Be aware that this function runs every re-render, even when it shouldn't (clicking on 'ask' button).

Bottom line: call this inside the useMemo

File: pages/asks.jsx [hunk 46]
```diff
@@ -0,0 +1,54 @@
...
+					/>
+				);
+			}
+
+			return acc;
+		}, []);
+	}, [showOnlyLikedAsks]);
+
+	return (
+		<main className={styles.main}>
```
Comment (liogiladi) [inline pages/asks.jsx:41]: In my opinion, would be more descriptive to identify all the pages' main element as `<page-name>-page` as its the root element/style of the page.

File: server/asksData.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (liogiladi) [reply server/asksData.js]: You misunderstood the comment - the problem was not with the `asksData` object's name. 
The problem was with the property `ask` in each object - this should be content (as it's the content of that specific ask)

[the remark this answers is not in the document]
Comment (liogiladi) [reply server/asksData.js]: A bit inconsistent with other booleans you've named (like isCurrentlyLiked).

File: styles/components/ask.module.scss [hunk 47]
```diff
@@ -0,0 +1,60 @@
...
+		cursor: pointer;
+
+		width: 1.75rem;
+		height: 1.75rem;
+		top: -0.5rem;
+		left: -0.5rem;
+
+		img {
+			width: 100%;
+			border-radius: 50%;
```
Comment (liogiladi) [inline styles/components/ask.module.scss:47]: wouldn't giving the border-radius to the a element also do the trick?

**Review by liogiladi — CHANGES_REQUESTED** [review]
a bit more

File: pages/asks.jsx [hunk 48]
```diff
@@ -0,0 +1,54 @@
+import { useState, useMemo } from "react";
+import styles from "@/styles/pages/asks.module.scss";
+import Ask from "@/components/Ask";
+import content from "@/server/content";
```
[the remark this answers is not in the document]
Comment (alonpeled44) [reply pages/asks.jsx:3]: noted

File: components/Modal.jsx (hunk 42 above)
Comment (alonpeled44) [reply components/Modal.jsx:13]: noted

File: components/Modal.jsx (hunk 43 above)
Comment (alonpeled44) [reply components/Modal.jsx]: noted

Comment (alonpeled44) [reply components/Modal.jsx]: noted

File: pages/asks.jsx (hunk 45 above)
Comment (alonpeled44) [reply pages/asks.jsx]: noted

File: pages/asks.jsx (hunk 46 above)
Comment (alonpeled44) [reply pages/asks.jsx:40]: I believe it's better the way it is right now, it clarify that you style the main content of the page, and not part of it.

File: styles/components/ask.module.scss (hunk 47 above)
Comment (alonpeled44) [reply styles/components/ask.module.scss]: noted

File: components/Modal.jsx [hunk 49]
```diff
@@ -0,0 +1,34 @@
...
+	};
+
+	const handleButtonClicked = (event) => {
+		if (event.key === "Escape") {
+			setIsOpen(false);
+		}
+	};
+
+	if (!isOpen) {
+		return;
```
Comment (liogiladi) [inline components/Modal.jsx:26]: I haven't noticed it and I know we talked about this subject 😅
but... consider this >w<
```js
if (!isOpen) return;
```

**Review by liogiladi — APPROVED** [review]
just added a minor pedantic suggestion
but we are good to go/for merge

File: components/Modal.jsx (hunk 49 above)
Comment (alonpeled44) [reply components/Modal.jsx]: noted
